libcxx/include/__availability

```diff
@@ -293,4 +304,11 @@
 #   define _LIBCPP_AVAILABILITY_THROW_BAD_VARIANT_ACCESS  _LIBCPP_AVAILABILITY_BAD_VARIANT_ACCESS
 #endif
 
+// Define availability attributes that depend on both 
+// _LIBCPP_HAS_NO_EXCEPTIONS and _LIBCPP_HAS_NO_RTTI.
+#if defined(_LIBCPP_HAS_NO_EXCEPTIONS) || defined(_LIBCPP_HAS_NO_RTTI)
+#   undef _LIBCPP_AVAILABILITY_HAS_INIT_PRIMARY_EXCEPTION
+#   define _LIBCPP_AVAILABILITY_HAS_INIT_PRIMARY_EXCEPTION 0
+#endif
+
```

Comment:
This hunk seems unnecessary to me right now, no? We could punt on adding it and let vendors define their availability attributes correctly based on whether exceptions are enabled or not.

Reply:
The idea is to simplify the process for vendors: no matter how they define `_LIBCPP_AVAILABILITY_HAS_INIT_PRIMARY_EXCEPTION` it just doesn't work without `RTTI` (on an unrelated note i believe it actually works with `NO_EXCEPTIONS`).